### PR TITLE
[IIDM V1.2] Add a convenient function to get a Terminal from its node (#67+#75)

### PR DESCRIPTION
**Pull request by sebalaig**
Signed-off-by: Sébastien LAIGRE <slaigre@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#67 and #75 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: include/powsybl/iidm/VoltageLevelViews.hpp [hunk 1]
```diff
@@ -129,6 +129,8 @@ class NodeBreakerView {
 
     virtual stdcxx::const_range<unsigned long> getNodes() const = 0;
 
+    virtual stdcxx::Reference<Terminal> getOptionalTerminal(unsigned long node) const = 0;
```
Comment (mathbagu) [inline include/powsybl/iidm/VoltageLevelViews.hpp:132]: You have to create two variants of this method, one `const` that returns a `CReference` and one no-const that returns a `Reference`

File: include/powsybl/math/UndirectedGraph.hxx [hunk 2]
```diff
@@ -450,6 +450,14 @@ void UndirectedGraph<V, E>::traverse(unsigned long v, const Traverser& traverser
     }
 }
 
+template <typename V, typename E>
+bool UndirectedGraph<V, E>::vertexExists(unsigned long v) const {
+    if (v >= m_vertices.size()) {
+        throw PowsyblException(stdcxx::format("Invalid vertex %1%", v));
+    }
+    return static_cast<bool>(m_vertices[v]);
+}
```
Comment (mathbagu) [inline include/powsybl/math/UndirectedGraph.hxx:459]: This is not consistent with the java code: In java, if `v` is greater or equal to `m_vertices.size()`, it returns false, it doesn't throw an exception.
As C++ makes lazy evaluation, you can simply copy/paste the java code

Comment (sebalaig) [reply include/powsybl/math/UndirectedGraph.hxx:456]: `v` is an `unsigned long` so the method cannot throw any exception unlike in Java.

File: include/powsybl/iidm/VoltageLevelViews.hpp (hunk 1 above)
Comment (sebalaig) [reply include/powsybl/iidm/VoltageLevelViews.hpp]: Ok, done

File: include/powsybl/math/UndirectedGraph.hxx (hunk 2 above)
Comment (mathbagu) [reply include/powsybl/math/UndirectedGraph.hxx:456]: Agree, but in your first version you would throw an exception if `v` is greater than the vertices size. This was my point. The java code, in that specific case returned false. I see that in the last version you fix it.

**Pull request merged**